pkg/metrics/query.go

```diff
@@ -0,0 +1,218 @@
+	}
+	client := &http.Client{Transport: tr}
+
+	req, err := http.NewRequest("GET", pq.URL, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Authorization", pq.BearerToken)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("curl command failed: %v\n%v", err, resp)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	return body, nil
+}
+
+// RunPrometheusQuery() checks that query returns non-empty result, errors if query returns empty
+// example: query := `ALERTS{alertstate="pending",alertname="PodDisruptionBudgetAtLimit",severity="warning"} == 1`
+func (pq *PromQuery) RunPrometheusQuery() error {
+	// expect all correct metrics within a reasonable time period
+	err := wait.Poll(time.Second*5, time.Second*waitForPrometheusStartSeconds, func() (bool, error) {
+		//TODO when the http/query apis discussed at https://github.com/prometheus/client_golang#client-for-the-prometheus-http-api
+		// and introduced at https://github.com/prometheus/client_golang/blob/master/api/prometheus/v1/api.go are vendored into
+		// openshift/origin, look to replace this homegrown http request / query param with that API
+		contents, err := pq.getBearerTokenURLViaPod()
+		if err != nil {
+			return false, fmt.Errorf("waitng for query: %s results", pq.Query)
+		}
+		result := prometheusResponse{}
+		json.Unmarshal(contents, &result)
+		metrics := result.Data.Result
+
+		if len(metrics) == 0 {
+			return false, fmt.Errorf("promQL query: %s had reported incorrect results: %v", pq.Query, metrics)
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (pq *PromQuery) tlsConfig() (*tls.Config, error) {
+	caBundle, err := pq.getCAbundle()
+	if err != nil {
+		return nil, err
+	}
+	certPool := x509.NewCertPool()
+	certPool.AppendCertsFromPEM(caBundle)
+	tlsConfig := &tls.Config{
+		RootCAs: certPool,
+	}
+	return tlsConfig, nil
+}
+
+func (pq *PromQuery) getCAbundle() ([]byte, error) {
+	// Prompt the injection of the ca bundle into a configmap
+	err := pq.createAnnotatedCABundleInjectionConfigMap()
+	if err != nil {
+		return nil, fmt.Errorf("error creating annotated configmap: %v", err)
+	}
+
+	// Retrieve the ca bundle
+	caBundlePEM, err := pq.pollForCABundle()
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving ca bundle: %v", err)
+	}
+	return caBundlePEM, nil
+
+}
+
+func (pq *PromQuery) createAnnotatedCABundleInjectionConfigMap() error {
+	_, err := pq.Client.CoreV1().ConfigMaps(pq.NsName).Create(&corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pq.ConfigMapName,
+			Annotations: map[string]string{
+				serviceCAinjectCABundleAnnotationName: "true",
+			},
+		},
+	})
+	return err
+}
+
+// pollForCABundle returns the given configmap data
+func (pq *PromQuery) pollForCABundle() ([]byte, error) {
+	expectedDataSize := 1
+	var ca []byte
+	err := wait.Poll(time.Second*5, time.Second*300, func() (bool, error) {
```

Comment:
Is there a reason not to use shorter intervals - 1s and 30s for example?

Reply:
nope, shortened, thnks